src/syft/lib/python/complex.py

```diff
@@ -1,25 +1,203 @@
 # stdlib
 from typing import Any
+from typing import List
 from typing import Optional
 
+# third party
+from google.protobuf.reflection import GeneratedProtocolMessageType
+
 # syft relative
+from ... import deserialize
+from ... import serialize
 from ...core.common import UID
+from ...core.store.storeable_object import StorableObject
 from ...decorators import syft_decorator
+from ...proto.lib.python.complex_pb2 import Complex as Complex_PB
+from ...util import aggressive_set_attr
+from .primitive_factory import PrimitiveFactory
 from .primitive_interface import PyPrimitive
 
 
 # TODO - actually make all of this work
 class Complex(complex, PyPrimitive):
     @syft_decorator(typechecking=True, prohibit_args=False)
-    def __new__(self, value: Any = None, id: Optional[UID] = None) -> "Complex":
-        if value is None:
-            value = 0.0
-        return complex.__new__(self, value)  # type: ignore
+    def __new__(
+        self, real: Any = None, imag: Any = None, id: Optional[UID] = None
+    ) -> "Complex":
+        if real is None:
+            return complex.__new__(self)
+        if imag is None:
+            return complex.__new__(self, real=real)  # type: ignore
+        if isinstance(real, str):
+            raise TypeError("Complex() can't take second arg if first is a string")
+        return complex.__new__(self, real=real, imag=imag)  # type: ignore
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __init__(self, real: Any = None, imag: Any = None, id: Optional[UID] = None):
+        complex.__init__(self)
+        self._id = id or UID()
+
+    @property
+    def id(self) -> UID:
+        """We reveal PyPrimitive.id as a property to discourage users and
+        developers of Syft from modifying .id attributes after an object
+        has been initialized.
+
+        :return: returns the unique id of the object
+        :rtype: UID
+        """
+        return self._id
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __add__(self, x: complex) -> "Complex":
+        result = complex.__add__(self, x)
+        return PrimitiveFactory.generate_primitive(value=result)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __sub__(self, x: complex) -> "Complex":
+        result = complex.__sub__(self, x)
+        return PrimitiveFactory.generate_primitive(value=result)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __mul__(self, x: complex) -> "Complex":
+        result = complex.__mul__(self, x)
+        return PrimitiveFactory.generate_primitive(value=result)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __pow__(self, x: complex) -> "Complex":
+        result = complex.__pow__(self, x)
+        return PrimitiveFactory.generate_primitive(value=result)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
```

Comment:
Q: ```prohibit_args=True```? (also for the other ones that we can :D

Reply:
Hum, maybe? Should we also set it to True for all our others types then?

Reply:
Hmm - I think I set it for the String type.
What do you think? @tudorcebere @Jasopaum @madhavajay @iamtrask 

My opinion is that is better to have them more restrictive to catch unwanted scenarios.